lib/MooseX/App/Meta/Role/Class/Base.pm

```diff
@@ -656,15 +656,18 @@ sub command_usage_options {
     $metaclass ||= $self;
     
     my @options;
-    foreach my $attribute ($self->command_usage_attributes($metaclass,[qw(option proto)])) {
+    foreach my $attribute (
+        sort {
+            $a->cmd_position <=> $b->cmd_position ||
+            $a->cmd_usage_name cmp $b->cmd_usage_name
+        } $self->command_usage_attributes($metaclass,[qw(option proto)])
+    ) {
         push(@options,[
             $attribute->cmd_usage_name(),
             $attribute->cmd_usage_description()
         ]);
     }
     
```

Comment:
Removing the sort at line 667 makes sense, however I don't think that you need to sort in the foreach loop since command_usage_attributes already returns a sorted list. Maybe add the command name sorting fallback there? ( ' || $a->cmd_usage_name cmp $b->cmd_usage_name' )